immuni_common/models/mongoengine/batch_file_eu.py

```diff
@@ -0,0 +1,132 @@
+#  Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#  Please refer to the AUTHORS file for more information.
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as
+#  published by the Free Software Foundation, either version 3 of the
+#  License, or (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU Affero General Public License for more details.
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import logging
+from datetime import date, datetime, timedelta
+from typing import Dict, List, Optional, Tuple
+
+from bson import ObjectId
+from mongoengine import BinaryField, DateTimeField, Document, EmbeddedDocumentListField, IntField, StringField
+
+from immuni_common.core.exceptions import NoBatchesException
+from immuni_common.models.mongoengine.temporary_exposure_key import TemporaryExposureKey
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BatchFileEu(Document):
```

Comment:
If both the already present and the EU batch classes have mostly the same underlying fields and functionality, I'd create a common superclass: this implementation goes against DRY. 🙏 
In the long run, it makes it difficult to maintain.

Reply:
This is the main comment to be addressed, IMHO.